frontend/apps/filemanager/filemanager.lua

```diff
@@ -1098,8 +1089,34 @@ function FileManager:createFolder()
     input_dialog:onShowKeyboard()
 end
 
-function FileManager:deleteFile(file)
-    local ok, err, is_dir
+function FileManager:deleteFileDialog(file, post_delete_callback, pre_delete_callback)
+    local file_abs_path = BaseUtil.realpath(file)
+    local is_file = lfs.attributes(file_abs_path, "mode") == "file"
+    local was_opened = is_file and DocSettings:hasSidecarFile(file_abs_path)
+    local text
+    if is_file then
+        text = _("Delete file permanently?")
+        if was_opened then
+            text = text .. _("\nBook settings, highlights and notes will be deleted.")
```

Comment:
I'm not too fond of the way this looks in the screenshot, but more generically, try to avoid bothering translators with things like newlines and spaces if it can be easily avoided. In this case:

```suggestion
            text = text .. "\n" .. _("Book settings, highlights and notes will be deleted.")
```